nano/secure/ledger.cpp

```diff
@@ -563,8 +539,10 @@ void ledger_processor::receive_block (nano::receive_block & block_a)
 			result.code = block_a.valid_predecessor (*previous) ? nano::process_result::progress : nano::process_result::block_position;
 			if (result.code == nano::process_result::progress)
 			{
-				auto account (ledger.store.frontier.get (transaction, block_a.hashables.previous));
-				result.code = account.is_zero () ? nano::process_result::gap_previous : nano::process_result::progress; // Have we seen the previous block? No entries for account at all (Harmless)
+				auto account = ledger.account (*previous);
+				auto info = ledger.account_info (transaction, account);
+				debug_assert (info);
+				result.code = info->head != block_a.hashables.previous ? nano::process_result::gap_previous : nano::process_result::progress; // Have we seen the previous block? No entries for account at all (Harmless)
```

Comment:
This does not seem equivalent to the previous check:
```
result.code = account.is_zero () ? nano::process_result::gap_previous : nano::process_result::progress; // Have we seen the previous block? No entries for account at all (Harmless)
```
If account info head is not empty and does not point to the previous hash, isn't it a fork?

Reply:
The check for `gap_previous` is done a few lines above:
```
result.code = previous != nullptr ? nano::process_result::progress : nano::process_result::gap_previous;
```

Reply:
Fixed this, the fork checking is now done in a similar fashion to other blocks.